data/json/furniture_and_terrain/appliances.json

```diff
@@ -968,6 +968,44 @@
     "damage_reduction": { "all": 60 },
     "variants": [ { "symbols": "0", "symbols_broken": "#" } ]
   },
+  {
+    "type": "vehicle_part",
+    "id": "ap_active_backup_generator",
+    "name": { "str": "active backup generator" },
+    "looks_like": "f_active_backup_generator",
+    "categories": [ "energy" ],
+    "color": "green_white",
+    "broken_color": "blue",
+    "location": "structure",
+    "damage_modifier": 80,
+    "durability": 400,
+    "description": "A ubiquitous piece of compact machinery meant for running anything from the bathroom lights to electric stoves and other necessities when grid electricity is unavailable.",
+    "//1": "BIZARRE FIELD NAMES HERE! Weird stuff happening! Epower determines how fast it is converted, power determines the ratio of diesel(mL) to battery charges output. In other words, *power* is *fuel efficiency*.",
+    "//2": "On a normal vehicle 250mL of diesel in an I4 with a truck alternator (what the active backup gen uses) should provide ~1692 charges (kW?) of power in about 24 minutes based on a test setup. That's a little less than 7W power, rounded up as I assume a larger dedicated generator is more efficient.",
+    "power": "7 W",
```

Comment:
This is a typo, should be 7 kw at least ( also for the comment above ), otherwise this appliance will drink up 20L of diesel in less than 2 minutes. According to https://en.wikipedia.org/wiki/Diesel_generator and https://www.generatorsource.com/Diesel_Fuel_Consumption.aspx , I think the reasonable number is around 10 kw.

Reply:
I'm not sure I understand. Increasing the `power` field will result in *generating more energy for a given amount of diesel*. (Yes that's weird!)

Using the table provided in the 2nd link, a 20kW generator is expected to use 1.6gal/hr (~6.05L). We're using about 625mL/hr (~10.3% of that), and our expected power output would thus be about 2.06kW.

Our actual power output, as defined by epower, is *1320 W*(1.32kW). So, we're off by a decent amount but it perfectly mirrors an existing ingame setup (the truck alternator+i4 diesel engine in a static frame).

We're certainly not off by *three orders of magnitude*, though?

Reply:
This is a math problem ^^. With `power: 7 W`, You need 188ml of diesel to generate 1320 joule of energy. So 188ml of diesel for one second! I'm not complaining about "power in the physical sense", but is really the matter of fuel efficiency.

Reply:
> So 188ml of diesel for one second!

This is what I am not understanding. What do you mean 'for one second'?

So, we have battery charges. Those are... *kilojoules*, yeah? (I know my comment says kW, power units are a bane!). And 1 kJ == 1kW*s (one kilowatt-second)?

188mL of diesel should produce about ~1272 charges (kJ) of battery according to the tests I did.

Reply:
> Oh probably I misunderstood. I'll check it and respond soon. Sorry.

Nothing to apologize for, please do let me know if you find an error. This stuff is confusing at the best of times, and I am not the best at writing comments for it!

Reply:
Ah I know what I got wrong. I took it for granted that "power": "7 W" means 1ml of diesel will produce 7 joules of energy. But in fact that's 7 kilo-joules. The unit of it made me dizzy. Sorry for wasting your time 😢

Reply:
Again, no worries!

Reply:
There's an existing vehicle called "portable_generator" that uses a 7.5kW generator instead of an alternator and a way smaller 0.6L V2 gasoline engine. I'm going to assume that the values they use are considered balanced and accurate so I'd suggest using those numbers for this generator, or at least a more powerful version.
Reason being that the 7.5kW generator requires 10444 engine power and the small V2 engine already provides 37300 of it.
Meanwhile a truck alternator requires 2238 engine power and both versions of the V4 inline produce 93250 power while consuming over twice the fuel. That's quite the large disparity in engine power provided versus necessary.
There's two even smaller 1-cylinder engines (25cc and 50cc) that could be used for smaller, weaker generators with car and truck alternators. Though the smallest engine alone could power multiples of them, if we're to follow the numbers.
Only issue is that all 3 engines I mentioned are gasoline only. Having diesel versions (for appliance generators) would be nice, although I'm not sure if 1-cylinders are available in diesel. The 0.6 V2 does happen to have real-world diesel counterparts.

Reply:
> There's an existing vehicle called "portable_generator" that uses a 7.5kW generator instead of an alternator and a way smaller 0.6L V2 gasoline engine. I'm going to assume that the values they use are considered balanced and accurate so I'd suggest using those numbers for this generator, or at least a more powerful version.
> 
> Reason being that the 7.5kW generator requires 10444 engine power and the small V2 engine already provides 37300 of it.
> 
> Meanwhile a truck alternator requires 2238 engine power and both versions of the V4 inline produce 93250 power while consuming over twice the fuel. That's quite the large disparity in engine power provided versus necessary.
> 
> There's two even smaller 1-cylinder engines (25cc and 50cc) that could be used for smaller, weaker generators with car and truck alternators. Though the smallest engine alone could power multiples of them, if we're to follow the numbers.
> 
> Only issue is that all 3 engines I mentioned are gasoline only. Having diesel versions (for appliance generators) would be nice, although I'm not sure if 1-cylinders are available in diesel. The 0.6 V2 does happen to have real-world diesel counterparts.

Yeah, this has been brought up in devcord too. It provides so little power for quite a large engine used in it, ESPECIALLY when compared to already existing Portable Generator.